### PR TITLE
Drop unused categories in `ExperimentAxisQuery.to_anndata`

### PR DESCRIPTION
**Pull request by nguyenv**
**Issue:**
https://github.com/single-cell-data/TileDB-SOMA/issues/2765

**Changes:**
`ExperimentAxisQuery.to_anndata` iterates through each column and drops unused categories.

**Notes for Reviewer:**
This change was originally proposed in https://github.com/single-cell-data/TileDB-SOMA/pull/2811 but was suggested somacore was a more appropriate location for it.

### CODE REVIEW

**Review by johnkerl — CHANGES_REQUESTED** [review]
From the discussion here

https://github.com/single-cell-data/TileDB-SOMA/issues/2765#issuecomment-2248061969

we need a new flag on `to_anndata`:

* optional
* default if omitted: current behavior
* if supplied and `True`: do what this PR does
* name: `droplevels`? Or something else? Let's chat with @pablo-gar @mojaveazure @eddelbuettel

File: python-spec/src/somacore/query/query.py [hunk 1]
```diff
@@ -298,7 +299,7 @@ def to_anndata(
```
Comment (johnkerl) [inline python-spec/src/somacore/query/query.py:299]: Let's add a description of what `drop_levels` does ...

Comment (johnkerl) [reply python-spec/src/somacore/query/query.py:299]: @pablo-gar @eddelbuettel @mojaveazure are you okay with `drop_levels`?

Comment (johnkerl) [reply python-spec/src/somacore/query/query.py:299]: (Presuming we'll want a same-ish name in Python & R both ...)

Comment (mojaveazure) [reply python-spec/src/somacore/query/query.py:302]: Seurat uses `drop` to match R, but I'm fine with `drop_levels` in SOMA

File: python-spec/src/somacore/query/query.py [hunk 2]
```diff
@@ -295,10 +296,13 @@ def to_anndata(
                 Additional varm layers to read and return in the varm slot.
             varp_layers:
                 Additional varp layers to read and return in the varp slot.
+            drop_levels:
+                Indicate whether unused categories on axis frames should be
+                dropped. By default, False, the categories are not dropped.
```
Comment (johnkerl) [inline python-spec/src/somacore/query/query.py:301]:
```suggestion
                dropped. By default, False, the categories which are present in the SOMA Experiment
                and not present in the query output are not dropped.
```

**Review by johnkerl — APPROVED** [review]
🚢 with a suggested docstring update

File: python-spec/src/somacore/query/query.py (hunk 1 above)
Comment (pablo-gar) [reply python-spec/src/somacore/query/query.py:303]: Sounds good to me!

**Pull request merged**